### PR TITLE
chez-matchable: init at 1.0

### PR DESCRIPTION
**Pull request by jitwit**
###### Motivation for this change

Adding a package definition for the hygenic pattern matching library developed by Alex Shinn.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/chez-modules/chez-matchable/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchgit, chez }:
+
+stdenv.mkDerivation {
+  pname = "chez-matchable";
+  version = "1.0";
+
+  src = fetchgit {
```
Comment (Lassulus) [inline pkgs/development/chez-modules/chez-matchable/default.nix]: maybe use `fetchFromGitHub` here

Comment (jitwit) [reply pkgs/development/chez-modules/chez-matchable/default.nix]: What's the difference?

Comment (Lassulus) [reply pkgs/development/chez-modules/chez-matchable/default.nix]: fetchFromGitHub fetches the tarball from github, which makes downloads faster

File: pkgs/development/chez-modules/chez-matchable/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    url = "https://github.com/fedeinthemix/chez-matchable.git";
+    rev = "73e46432ae70ec72eba6ef116bd84ad9ee38b2f2";
```
Comment (doronbehar) [inline pkgs/development/chez-modules/chez-matchable/default.nix]: Also, please use the version as the `rev` - the Git tag.

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/development/chez-modules/chez-matchable/default.nix (hunk 1 above)
Comment (jitwit) [reply pkgs/development/chez-modules/chez-matchable/default.nix]: Sorry for the delay, done! (assuming it looks ok now?)

**Review by Lassulus — APPROVED** [review]
builds with nix-review, LGTM

EDIT: wait, the version is not 1.0 anymore, can you change the commit message?
EDIT EDIT: ah nvm, I can change it when squasing

**Comment by jitwit** [comment]
> builds with nix-review, LGTM
> 
> EDIT: wait, the version is not 1.0 anymore, can you change the commit message?
> EDIT EDIT: ah nvm, I can change it when squasing

Ach, sorry for missing that. Thanks!

**Pull request merged**